test/web_socket_task_test.rb

```diff
@@ -0,0 +1,254 @@
+# frozen_string_literal: true
+
+require 'kontena-websocket-client'
+require 'json'
+
+using_task_library 'controldev_websocket'
+
+import_types_from 'controldev_websocket'
+
+WebsocketStruct = Concurrent::MutableStruct.new :ws,
+                  :received_messages, :current_state, :has_error
+
+def websocket_connect(state)
+    Kontena::Websocket::Client.connect(@url) do |ws|
+        state.ws = ws
+        state.current_state = :open
+
+        ws.read do |message|
+            state.received_messages.append(message)
+        end
+    end
+rescue Kontena::Websocket::CloseError => exc
+    state.current_state = :closed
+rescue Kontena::Websocket::Error => exc
+    state.has_error = true
+end
+
+describe OroGen.controldev_websocket.Task do
+    run_live
+
+    before do
+        @task = task = syskit_deploy(
+            OroGen.controldev_websocket.Task.deployed_as('websocket')
+        )
+        task.properties.port = 65432
+        task.axis_map = [Mapping. new,
+                         Mapping. new,
+                         Mapping. new,
+                         Mapping. new,
+                         Mapping. new,
+                         Mapping. new]
+        task.axis_map[0].index = 0
+        task.axis_map[0].type = 0
+        task.axis_map[1].index = 1
+        task.axis_map[1].type = 0
+        task.axis_map[2].index = 6
+        task.axis_map[2].type = 1
+        task.axis_map[3].index = 2
+        task.axis_map[3].type = 0
+        task.axis_map[4].index = 3
+        task.axis_map[4].type = 0
+        task.axis_map[5].index = 7
+        task.axis_map[5].type = 1
+
+        task.button_map = [ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new,
+                           ButtonMapping. new]
+        thresh = 0.5
+        task.button_map[0].index = 0
+        task.button_map[2].index = 1
+        task.button_map[3].index = 2
+        task.button_map[4].index = 3
+        task.button_map[5].index = 4
+        task.button_map[6].index = 5
+        task.button_map[7].index = 8
+        task.button_map[8].index = 9
+        task.button_map[9].index = 12
+        task.button_map[10].index = 13
+        task.button_map[11].index = 14
+        task.button_map[12].index = 15
```

Comment:
You're not setting the `type` field. Also,
- should be `task.properties`
- can't change properties on-the-fly. Writing a property accessor is what causes an update, e.g. `task.properties.button_map = ` is what triggers an update.
- https://www.rock-robotics.org/rock-and-syskit/components/types_in_ruby.html